bucket/godot.json

```diff
@@ -13,14 +13,21 @@
             "hash": "5d99d41df6a9c46fb8da2c156df560e71a81c7d2a50bbc12ed210602d0f86a97"
         }
     },
-    "pre_install": "Get-ChildItem \"$dir\\Godot_*.exe\" | Rename-Item -NewName \"$dir\\godot.exe\"",
+    "pre_install": [
+        "Get-ChildItem \"$dir\\Godot_*.exe\" | Rename-Item -NewName \"$dir\\godot.exe\"",
+        "New-Item \"$dir\\_sc_\" -ItemType \"file\" | Out-Null"
+    ],
     "bin": "godot.exe",
     "shortcuts": [
         [
             "godot.exe",
             "Godot"
         ]
     ],
+    "persist": "editor_data",
+    "post_install": [
+        "if (!(Test-Path \"$persist_dir\\editor_data\\editor_settings-3.tres\") -and (Test-Path \"$Env:AppData\\Godot\\editor_settings-3.tres\")) {Copy-Item -Path \"$Env:AppData\\Godot\\*\" -Destination \"$persist_dir\\editor_data\" -Recurse -Force}"
+    ],
```

Comment:
- Move it directly after `pre_install`
- Why do you check for `editor_settings-3.tres`?
- `$Env` is defined as `$env`
- `$Env:AppData` == `$env:APPDATA`

Reply:
- Move `post_install` property **directly** after `pre_install` property
- Why did you check for editor_settings-3.tres?
- Why do you use `Measure-Object`?

Reply:
Q: Why did you check for editor_settings-3.tres?
A: No, I didn't use it this time :D

Q: Why do you use Measure-Object?
A: Testing for the folder is empty or not

Reply:
> Why did you check for editor_settings-3.tres?

Past tense. Why did you do it in first time??

> A: Testing for the folder is empty or not

Does not make sense... Absolutely wasted `measure-object` call when you can use direct array property.

![image](https://user-images.githubusercontent.com/13260377/90507710-9fa43f80-e156-11ea-9fa7-d2f110467d77.png)

Reply:
`if (((Get-ChildItem \"$persist_dir\\editor_data\" -Force | Select-Object -First 1 | Measure-Object).Count -eq 0) -and ((Get-ChildItem \"$env:APPDATA\\Godot\" -Force | Select-Object -First 1 | Measure-Object).Count -ne 0)) {Copy-Item -Path \"$env:APPDATA\\Godot\\*\" -Destination \"$persist_dir\\editor_data\" -Recurse -Force}`

Why did you ever think this is good approach??

`gci` calls on directory, always return array. Classic nature of native powershell cmdlets

So this code will do the following:

1. Gci call
	1. In case the directory do not exists, it will fail and throw exception
	1. -Force parameter does not make sense.
1. Limit the output of gci into 1 Item for no reason
1. Nonsensual `measure-object` call just for slowing down the execution
1. `Copy-Item` call
	1. It will produce junk in case the `editor_data` folder do not exists (first ever case on update)

Your approach of copying all files:

![image](https://user-images.githubusercontent.com/13260377/90508759-8ef4c900-e158-11ea-9fa8-e379c62dec76.png)


Correct:

![image](https://user-images.githubusercontent.com/13260377/90508704-797f9f00-e158-11ea-887c-3921d1b2170e.png)

Reply:
I don't know PowerShell. I just want to contribute to scoop repo.
I wasn't know how to test a folder is empty or not. Then, today I found a way in a forum.

> It will produce junk in case the editor_data folder do not exists

but scoop creates persist folder on update

without persist and post_install:
![Screenshot_16](https://user-images.githubusercontent.com/49201485/90512903-8a83dc80-e167-11ea-8877-a8e58b92b787.png)

with persist and post_install (fake update):
![Screenshot_17](https://user-images.githubusercontent.com/49201485/90512918-8eaffa00-e167-11ea-8aad-1d726398cbc8.png)

Reply:
Then create issue instead of posting non functional broken PR, when you do not know powershell...

Reply:
Forget it. I will do it on my own

Reply:
>:///